csvGenerator.ipynb

```diff
@@ -0,0 +1,409 @@
+    "import constant as const \n",
+    "from authObject import AuthObject\n",
+    "from CitesphereConnector import CitesphereConnector"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 20,
+   "id": "2c0597b1",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "auth_object = AuthObject()\n",
+    "auth_object.authType  = 'oauth'\n",
+    "auth_object.access_token=\"citesphere-access-token\"\n",
+    "citesphere_api_url = const.CITESPHERE_API_URL\n",
+    "connector = CitesphereConnector(citesphere_api_url, auth_object)\n",
+    "#default max number of items displayed on a collection items page in citesphere\n",
+    "max_size=const.MAX_SIZE"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "e56d14e7",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "\"\"\"\n",
+    "groups is a list of dictionaries contains information about the different groups of collections \n",
+    "with following properties:\n",
+    "\n",
+    "{'id': ,\n",
+    "  'name': '',\n",
+    "  'version': ,\n",
+    "  'created': <date>,\n",
+    "  'lastModified': <date>,\n",
+    "  'numItems': ,\n",
+    "  'owner': ,\n",
+    "  'type': '',\n",
+    "  'description': '',\n",
+    "  'url': '',\n",
+    "  'syncInfo': }\n",
+    "  \n",
+    "\"\"\""
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 34,
+   "id": "6dc3bede",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "groups=connector.get_groups()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "5c5a2298",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "\"\"\"\n",
+    "\n",
+    "this following function \"download_file(folder_path:str,ids:set, citesphere_token:str) -> list\":\n",
+    "\n",
+    "has following parameters:\n",
+    "\n",
+    "1. folder_path:str = location of the folder where the user wants to save the downloaded files\n",
+    "\n",
+    "2. ids:set = set of tuples, containing pair of file ids and file names\n",
+    "\n",
+    "3. citesphere_token:str = token to access data from giles\n",
+    "\n",
+    "function:\n",
+    "this function iterates throught the ids list of tuples, plugging the file id to the giles url string to get the\n",
+    "file url from giles for downloading the file to the dedicated folder.\n",
+    "\n",
+    "Append the path to the path_list.\n",
+    "\n",
+    "if the response of the get request is of level 200\n",
+    "it writes the file with the appriate content and save it to the download folder and *returns* the list of the path\n",
+    "for the downloaded files, so that it can be saved to the csv file's path field.\n",
+    "\n",
+    "\"\"\""
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 36,
+   "id": "d91846ec",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#download files from the collection items\n",
+    "def download_file(folder_path:str,ids:set, citesphere_token:str) -> list:\n",
```

Comment:
I think I commented on that before. If this function download multiple files, the name must indicate that (`files`).